### DISCUSSION TITLE
Alpine Virtual Machine as Self-Hosted Github Actions Runner

### DISCUSSION THREAD

**Discussion by flavius-t** [community/community#30593] [Actions]
Hi,

I'm running an Alpine Linux 3.16 VM which I'm trying to set up as a self-hosted runner for Actions. I'm following the instructions listed for adding a Linux x64 runner but under the Configure section, trying to run `./config.sh --url https://github.com/<org>/<repo/ --token <token>` gives me errors:

```
Error loading shared library ld-linux-x86-64.so.2: No such file or directory (needed by ./bin/libcoreclr.so)
Error relocating ./bin/libcoreclr.so: __isnan: symbol not found
Error relocating ./bin/libcoreclr.so: __isnanf: symbol not found
Libicu's dependencies is missing for Dotnet Core 6.0
Execute sudo ./bin/installdependencies.sh to install any missing Dotnet Core 6.0 dependencies.
```

Trying to execute `sudo ./bin/installdependencies.sh` gives this:

```
Can't detect current OS type based on /etc/os-release.
Can't install dotnet core dependencies.
You can manually install all required dependencies based on following documentation
https://docs.microsoft.com/en-us/dotnet/core/linux-prerequisites?tabs=netcore2x
```

I've tried manually installing the dotnet core dependencies for Alpine listed here: https://docs.microsoft.com/en-us/dotnet/core/install/linux-alpine, and then manually installing .NET6 SDK. No luck, still getting the same errors for both of the above. I see there was a [previous discussion](https://github.com/actions/runner/issues/801) similar to this issue, but no clear resolution. 

Can this be done, or is Alpine not supported as a self-hosted runner?

**Comment by naikrovek** [comment]
most of the time, Actions assumes Ubuntu, if you’re using Linux.  There is no way that every flavor of Linux can be supported by the folks maintaining this, or even most of the major Linux distros.

if you have any error messages or logs that came up during the .net 6 prerequisite installation, those would help work out what’s happening, probably.  if there were any errors, that is.

if you havent done so already, try installing the `gcompat` package for alpine, and see if that gets you further along.
